src/mlpack/methods/ann/activation_functions/elish_function.hpp

```diff
@@ -0,0 +1,122 @@
+/**
+ * @file elish_function.hpp
+ * @author Bisakh Mondal
+ *
+ * Definition and implementation of the ELiSH function as described by
+ * Mina Basirat and Peter M. Roth.
+ *
+ * For more information see the following paper
+ * 
+ * @code
+ * @misc{ELiSHActivation,
+ *    title={The Quest for the Golden Activation Function},
+ *    author={Mina Basirat and Peter M. Roth},
+ *    year={2018},
+ *    url={https://arxiv.org/pdf/1808.00783.pdf},
+ *    eprint={1808.00783},
+ *    archivePrefix={arXiv},
+ *    primaryClass={cs.NE} }
+ * @endcode
+ * 
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_ELISH_FUNCTION_HPP
+#define MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_ELISH_FUNCTION_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The ELiSH function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) &=& \left\{
+ *   \begin{array}{lr}
+ *     \frac{x}{1+e^{-x}} & : x \ge 0 \\
+ *     \frac{e^x-1}{1+e^{-x}} & : x < 0
+ *   \end{array}
+ * \right \\
+ * f'(x) &=& \left\{
+ *   \begin{array}{lr}
+ *     \frac{1}{1+e^{-x}} + 
+ *              \frac{xe^{-x}}{ \left(1+e^{-x}}\right)^2} & : x \ge 0 \\
```

Comment:
Hey any reason why we need to do indent this. Let me know what you think. Thanks.

Reply:
That's a mistake. Thanks. Changing.